picard/ui/options/interface.py

```diff
@@ -227,16 +237,16 @@ def save(self):
         config.setting["use_adv_search_syntax"] = self.ui.use_adv_search_syntax.isChecked()
         config.setting["quit_confirmation"] = self.ui.quit_confirmation.isChecked()
         self.tagger.window.update_toolbar_style()
-        new_theme_setting = self.ui.use_system_theme.isChecked()
+        new_theme_setting = self.ui.ui_theme.itemData(self.ui.ui_theme.currentIndex())
         new_language = self.ui.ui_language.itemData(self.ui.ui_language.currentIndex())
         restart_warning = None
-        if new_theme_setting != config.setting["use_system_theme"]:
+        if new_theme_setting != config.setting["ui_theme"]:
             restart_warning_title = _('Theme changed')
             restart_warning = _('You have changed the application theme. You have to restart Picard in order for the change to take effect.')
             if new_theme_setting:
                 restart_warning += '\n\n' + _(
                     'Please note that using the system theme might cause the user interface to be not shown correctly. '
-                    'If this is the case disable the "Use system theme" option to use Picard\'s default theme again.'
+                    'If this is the case select the "Light" theme option to use Picard\'s default theme again.'
```

Comment:
Under Linux, one only has choice between `System` or `Default`, so this message doesn't have much sense, as there's no `Light` them option to select:
![image](https://user-images.githubusercontent.com/151042/111074844-5007d580-84e5-11eb-8a0e-82a97878405c.png)

Reply:
🧐
Should I change the Windows/Mac DEFAULT_THEME option to SYSTEM_THEME, make it the default value and rename the DEFAULT_THEME to FUSION_THEME?

After that I would change the "Light" in this message to "Fusion" (or "Default").

Reply:
I think we are mixing up things here. There are both Qt5 themes and color palettes. So what we have is the following:

- Windows: We always use the "Fusion" theme, no exception here. By default we auto detect if Windows is configured for light or dark UI. In light we use Fusion's default color palette, in dark we use a custom one. We need the user be able to select between three states: "System", "Dark" and "Light".
- macOS: We always use Qt's macOS theme, no exception. In theory and when running from source we can detect whether light or dark UI is used. In the current app package this does not work. But we can explicitly force dark or light UI. We need the user be able to select between three states: "Default", "Dark" and "Light"
- Linux and most other OS: By default we use the "Fusion" theme, but we allow users to opt-in to use the system configured Qt5 theme. In both cases we can end up with light and dark UI (e.g. on KDE with dark UI configured it will set a dark color palette for Qt5 independent of the theme). We detect this via a heuristic and adapt some UI colors accordingly. We need two options: "Default" (using the Fusion theme) and "System"
- Haiku: We have some small special case here. We always use the default Haiku theme. There is no explicit dark or light UI option in Haiku. The user might still have configured dark UI colors. We detect this via the heuristic. We need no option for the user.

Reply:
> The Windows/Mac "Default" theme changes with the system theme (like the Linux "System" theme).

Just to clarify: This is not exactly the same. We always use Fusion on Windows and we always use the mac theme macOS, because those platforms do not have by default any way for the user to change their Qt5 theme. Linux on the other hand does, but by default we force the Fusion theme.